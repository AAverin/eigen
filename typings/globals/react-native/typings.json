{
  "resolution": "main",
  "tree": {
    "src": "https://raw.githubusercontent.com/DefinitelyTyped/DefinitelyTyped/4c7d03b088d4e71cd9fe9c66ab7bd363049077c3/react-native/react-native.d.ts",
    "raw": "registry:dt/react-native#0.29.0+20160719232534",
    "typings": "https://raw.githubusercontent.com/DefinitelyTyped/DefinitelyTyped/4c7d03b088d4e71cd9fe9c66ab7bd363049077c3/react-native/react-native.d.ts"
  }
}
