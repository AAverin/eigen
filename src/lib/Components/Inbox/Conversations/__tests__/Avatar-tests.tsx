import "jest-snapshots-svg"
import * as React from "react"
import "react-native"
import * as renderer from "react-test-renderer"

import Avatar from "../Avatar"

it("looks correct for a user with two initials", () => {
  const avatar = renderer.create(<Avatar isUser={true} senderName={"Maxim Cramer"} />).toJSON()
  expect(avatar).toMatchSnapshot()
})

it("looks correct for a user with three initials", () => {
  const avatar = renderer.create(<Avatar isUser={true} senderName={"Amazing Person Name"} />).toJSON()
  expect(avatar).toMatchSnapshot()
})

it("looks correct for a user with many initials", () => {
  const avatar = renderer
    .create(<Avatar isUser={true} senderName={"Amazing Person With Very Very Long Name"} />)
    .toJSON()
  expect(avatar).toMatchSnapshot()
})

it("looks correct for a gallery with two initials", () => {
  const avatar = renderer.create(<Avatar isUser={false} senderName={"Best Gallery"} />).toJSON()
  expect(avatar).toMatchSnapshot()
})

it("looks correct for a gallery with many initials", () => {
  const avatar = renderer
    .create(<Avatar isUser={false} senderName={"Best Gallery Ever But Has Very Long Name"} />)
    .toJSON()
  expect(avatar).toMatchSnapshot()
})

it("looks correct for a gallery with lowercase initials", () => {
  const avatar = renderer.create(<Avatar isUser={false} senderName={"all lower case though"} />).toJSON()
  expect(avatar).toMatchSnapshot()
})
