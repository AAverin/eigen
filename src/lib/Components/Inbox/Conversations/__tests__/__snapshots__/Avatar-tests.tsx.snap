// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`looks correct for a gallery with lowercase initials 1`] = `
<View
  style={
    Array [
      Object {
        "backgroundColor": "#e5e5e5",
        "borderRadius": 15,
        "height": 30,
        "width": 30,
      },
      undefined,
    ]
  }
>
  <Text
    accessible={true}
    allowFontScaling={true}
    disabled={false}
    ellipsizeMode="tail"
    style={
      Array [
        Object {
          "alignSelf": "center",
          "color": "#666666",
          "fontFamily": "Avant Garde Gothic ITCW01Dm",
          "fontSize": 10,
          "marginTop": 8,
        },
        undefined,
      ]
    }
  >
    ALC
  </Text>
</View>
`;

exports[`looks correct for a gallery with many initials 1`] = `
<View
  style={
    Array [
      Object {
        "backgroundColor": "#e5e5e5",
        "borderRadius": 15,
        "height": 30,
        "width": 30,
      },
      undefined,
    ]
  }
>
  <Text
    accessible={true}
    allowFontScaling={true}
    disabled={false}
    ellipsizeMode="tail"
    style={
      Array [
        Object {
          "alignSelf": "center",
          "color": "#666666",
          "fontFamily": "Avant Garde Gothic ITCW01Dm",
          "fontSize": 10,
          "marginTop": 8,
        },
        undefined,
      ]
    }
  >
    BGE
  </Text>
</View>
`;

exports[`looks correct for a gallery with two initials 1`] = `
<View
  style={
    Array [
      Object {
        "backgroundColor": "#e5e5e5",
        "borderRadius": 15,
        "height": 30,
        "width": 30,
      },
      undefined,
    ]
  }
>
  <Text
    accessible={true}
    allowFontScaling={true}
    disabled={false}
    ellipsizeMode="tail"
    style={
      Array [
        Object {
          "alignSelf": "center",
          "color": "#666666",
          "fontFamily": "Avant Garde Gothic ITCW01Dm",
          "fontSize": 10,
          "marginTop": 8,
        },
        undefined,
      ]
    }
  >
    BG
  </Text>
</View>
`;

exports[`looks correct for a user with many initials 1`] = `
<View
  style={
    Array [
      Object {
        "backgroundColor": "#e5e5e5",
        "borderRadius": 15,
        "height": 30,
        "width": 30,
      },
      undefined,
    ]
  }
>
  <Text
    accessible={true}
    allowFontScaling={true}
    disabled={false}
    ellipsizeMode="tail"
    style={
      Array [
        Object {
          "alignSelf": "center",
          "color": "white",
          "fontFamily": "Avant Garde Gothic ITCW01Dm",
          "fontSize": 10,
          "marginTop": 8,
        },
        undefined,
      ]
    }
    user={true}
  >
    APW
  </Text>
</View>
`;

exports[`looks correct for a user with three initials 1`] = `
<View
  style={
    Array [
      Object {
        "backgroundColor": "#e5e5e5",
        "borderRadius": 15,
        "height": 30,
        "width": 30,
      },
      undefined,
    ]
  }
>
  <Text
    accessible={true}
    allowFontScaling={true}
    disabled={false}
    ellipsizeMode="tail"
    style={
      Array [
        Object {
          "alignSelf": "center",
          "color": "white",
          "fontFamily": "Avant Garde Gothic ITCW01Dm",
          "fontSize": 10,
          "marginTop": 8,
        },
        undefined,
      ]
    }
    user={true}
  >
    APN
  </Text>
</View>
`;

exports[`looks correct for a user with two initials 1`] = `
<View
  style={
    Array [
      Object {
        "backgroundColor": "#e5e5e5",
        "borderRadius": 15,
        "height": 30,
        "width": 30,
      },
      undefined,
    ]
  }
>
  <Text
    accessible={true}
    allowFontScaling={true}
    disabled={false}
    ellipsizeMode="tail"
    style={
      Array [
        Object {
          "alignSelf": "center",
          "color": "white",
          "fontFamily": "Avant Garde Gothic ITCW01Dm",
          "fontSize": 10,
          "marginTop": 8,
        },
        undefined,
      ]
    }
    user={true}
  >
    MC
  </Text>
</View>
`;
